### PR TITLE
Revert "Revert "Fix repeater report""

### PR DESCRIPTION
**Pull request by benrudolph**
Reverts dimagi/commcare-hq#15464

@proteusvacuum @mkangia should wait til weekend to merge

### CODE REVIEW

File: corehq/apps/locations/static/locations/js/location_drilldown.async.js [hunk 1]
```diff
@@ -1,27 +1,28 @@
...
     $('#loc_ajax').show().removeClass('hide');
-    $.getJSON(LOAD_LOCS_URL, params, function(allData) {
+    $.getJSON(loc_url, params, function(allData) {
         $('#loc_ajax').hide().addClass('hide');
         callback(allData.objects);
     });
 }
 
-function LocationSelectViewModel(hierarchy, default_caption, auto_drill, loc_filter, func, show_location_filter) {
+function LocationSelectViewModel(options) {
```
Comment (dimagimon) [inline corehq/apps/locations/static/locations/js/location_drilldown.async.js]: 'LocationSelectViewModel' is defined but never used (no-unused-vars)

File: corehq/apps/locations/static/locations/js/location_drilldown.async.js [hunk 2]
```diff
@@ -1,27 +1,28 @@
...
 
-    this.default_caption = default_caption || 'All';
-    this.auto_drill = (_.isBoolean(auto_drill) ? auto_drill : true);
-    this.loc_filter = loc_filter || function(loc) { return true; };
-    this.func = typeof func !== 'undefined' ? func : LocationModel;
-    this.show_location_filter = ko.observable((typeof show_location_filter !== 'undefined') ? show_location_filter : 'y');
+    this.loc_url = options.loc_url;
+    this.default_caption = options.default_caption || 'All';
+    this.auto_drill = (_.isBoolean(options.auto_drill) ? options.auto_drill : true);
+    this.loc_filter = options.loc_filter || function(loc) { return true; };
```
Comment (dimagimon) [inline corehq/apps/locations/static/locations/js/location_drilldown.async.js]: 'loc' is defined but never used (no-unused-vars)

File: corehq/apps/reports_core/templates/reports_core/filters/location_async/location_async.js [hunk 3]
```diff
@@ -0,0 +1,20 @@
+  {% load reports_core_tags %}
```
Comment (dimagimon) [inline corehq/apps/reports_core/templates/reports_core/filters/location_async/location_async.js]: Parsing error: Unexpected token %

**Comment by mkangia** [comment]
wow! what happened here.. quite many contributions!

**Comment by benrudolph** [comment]
heh accidentally merged this branch into another one that was caught up with latest head then pushed the changes

**Pull request closed without merging**